### PR TITLE
Storvsc backports

### PR DESCRIPTION
**Pull request by sharsonia**

### CODE REVIEW

**Comment by sharsonia** [comment]
@johnsongeorge-w

**Comment by johnsongeorge-w** [comment]
1. Limit the commit messages to 80 characters. Move the Backporting info to next line.
2. Specify the upstream link used for backporting. Refer the previous backports.

**Review by johnsongeorge-w — CHANGES_REQUESTED** [review]

File: hv-rhel6.x/hv/storvsc_drv.c [hunk 1]
```diff
@@ -2298,19 +2301,28 @@ static int storvsc_probe(struct hv_device *device,
 			 device->dev_instance.b[4]);
 		ret = scsi_add_device(host, 0, target, 0);
 		if (ret) {
```
Comment (johnsongeorge-w) [inline hv-rhel6.x/hv/storvsc_drv.c:2311]: Remove curly braces to match the backport

File: hv-rhel6.x/hv/storvsc_drv.c [hunk 2]
```diff
@@ -521,6 +521,8 @@ struct hv_host_device {
 	unsigned char target;
 	struct workqueue_struct *handle_error_wq;
 	struct mutex host_mutex;
+	struct work_struct host_scan_work;
+	struct Scsi_Host *host
```
Comment (johnsongeorge-w) [inline hv-rhel6.x/hv/storvsc_drv.c:525]: This code will not compile.

File: hv-rhel6.x/hv/storvsc_drv.c [hunk 3]
```diff
@@ -2289,10 +2291,19 @@ static int storvsc_probe(struct hv_device *device,
 		host->nr_hw_queues = stor_device->num_sc + 1;
 #endif
 
+	/*
+	 * Set the error handler work queue.
+	 */
+	host_dev->handle_error_wq =
+			alloc_ordered_workqueue("storvsc_error_wq_%d",
```
Comment (johnsongeorge-w) [inline hv-rhel6.x/hv/storvsc_drv.c:2294]: In 2.6.32-754.el6.x86_64, alloc_ordered_workqueue takes only 2 args.
/usr/src/kernels/2.6.32-754.el6.x86_64/include/linux/workqueue.h:#define alloc_ordered_workqueue(name, flags) create_singlethread_workqueue(name)

File: hv-rhel6.x/hv/storvsc_drv.c [hunk 4]
```diff
@@ -2297,20 +2300,28 @@ static int storvsc_probe(struct hv_device *device,
...
-			scsi_remove_host(host);
-			goto err_out2;
-		}
+		if (ret)
+			goto err_out3;
 	}
 #if defined(CONFIG_SCSI_FC_ATTRS) || defined(CONFIG_SCSI_FC_ATTRS_MODULE)
 	if (host->transportt == fc_transport_template) {
+		struct fc_rport_identifiers ids = {
+			.roles = FC_PORT_ROLE_FCP_DUMMY_INITIATOR,
```
Comment (johnsongeorge-w) [inline hv-rhel6.x/hv/storvsc_drv.c:2317]: FC_PORT_ROLE_FCP_DUMMY_INITIATOR is not defined.

**Review by johnsongeorge-w — APPROVED** [review]

**Pull request merged**